compactor/src/compact.rs

```diff
@@ -949,69 +1067,84 @@ impl Compactor {
+                        parquet_files[0].table_id(),
+                        overall_min_max_sequence_number,
+                        overall_min_time,
+                        overall_max_time,
+                    )
+                    .await
+                    .context(QueryingTombstonesSnafu)?;
+
+                let parquet_files = parquet_files
+                    .into_iter()
+                    .map(|data| {
+                        // Filter the set of tombstones relevant to any file in the group to just those
+                        // relevant to this particular parquet file.
+                        let relevant_tombstones = tombstones
+                            .iter()
+                            .cloned()
+                            .filter(|t| {
+                                t.sequence_number > data.max_sequence_number()
+                                    && ((t.min_time <= data.min_time()
+                                        && t.max_time >= data.min_time())
+                                        || (t.min_time > data.min_time()
+                                            && t.min_time <= data.max_time()))
+                            })
+                            .collect();
+
+                        ParquetFileWithTombstone {
+                            data: Arc::new(data.parquet_file()),
+                            tombstones: relevant_tombstones,
+                        }
+                    })
+                    .collect();
 
-            let parquet_files = parquet_files
-                .into_iter()
-                .map(|data| {
-                    // Filter the set of tombstones relevant to any file in the group to just those
-                    // relevant to this particular parquet file.
-                    let relevant_tombstones = tombstones
+                overlapped_file_with_tombstones_groups.push(GroupWithTombstones {
+                    parquet_files,
+                    tombstones,
+                });
+            } else {
+                // Already has tombstones attached to the parquet files, use them
+                let group = GroupWithTombstones::new_from_parquet_files_with_tombstones(
+                    parquet_files
                         .iter()
-                        .cloned()
-                        .filter(|t| {
-                            t.sequence_number > data.max_sequence_number
-                                && ((t.min_time <= data.min_time && t.max_time >= data.min_time)
-                                    || (t.min_time > data.min_time && t.min_time <= data.max_time))
-                        })
-                        .collect();
-
-                    ParquetFileWithTombstone {
-                        data: Arc::new(data),
-                        tombstones: relevant_tombstones,
-                    }
-                })
-                .collect();
+                        .map(|f| f.parquet_files_with_timbstone())
```

Comment:
Nit, typo in method name here:

```suggestion
                        .map(|f| f.parquet_files_with_tombstone())
```